spec/lib/blockchain_adapters/insight_adapter_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+
+RSpec.describe Straight::Blockchain::InsightAdapter do
+
+  subject(:mainnet_adapter) { Straight::Blockchain::InsightAdapter.mainnet("https://insight.mycelium.com/api") }
+
+  before :all do
+    VCR.insert_cassette 'blockchain_insight_adapter'
+  end
+
+  after :all do
+    VCR.eject_cassette
+  end
+
+  let(:tid) { 'b168b57a9ae38c0671c5eef3be6c8305782bd1351e75028dac491185388d5424' }
+
+  it "fetch a single transaction" do
+    expect(mainnet_adapter.fetch_transaction(tid)[:total_amount]).to eq(499900000)
+  end
+
+  it "return correct prepared data" do
+    expect(mainnet_adapter.fetch_transaction(tid)[:total_amount]).to eq(499900000)
+    expect(mainnet_adapter.fetch_transaction(tid)[:tid]).to eq("b168b57a9ae38c0671c5eef3be6c8305782bd1351e75028dac491185388d5424")
+    expect(mainnet_adapter.fetch_transaction(tid)[:outs].first[:amount]).to eq(187000000)
+  end
+
+  it "fetch first transaction for the given address"  do
+    address = "1CBWzY7PEnUtT4b36bth4UZuNmby9pTT7A"
+    expect(mainnet_adapter.fetch_transactions_for(address)).to be_kind_of(Array)
+    expect(mainnet_adapter.fetch_transactions_for(address)).not_to be_empty
+  end
+
+  it "fetch balance for given address" do
```

Comment:
fetchES and returnS and also raiseS.
Verb's third form with "It" :)
Please check everywhere.